mmv1/third_party/terraform/provider/provider_impersonate_service_account_delegates_test.go

```diff
@@ -0,0 +1,197 @@
+		// see https://github.com/golang/go/wiki/CommonMistakes#using-goroutines-on-loop-iterator-variables
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			tc(t)
+		})
+	}
+}
+
+func testAccSdkProvider_impersonate_service_account_delegates_setInConfig(t *testing.T) {
+	acctest.SkipIfVcr(t) // Test doesn't interact with API
+
+	delegates := []string{
+		"projects/-/serviceAccounts/my-service-account-1@example.iam.gserviceaccount.com",
+		"projects/-/serviceAccounts/my-service-account-2@example.iam.gserviceaccount.com",
+	}
+	delegatesString := fmt.Sprintf(`["%s","%s"]`, delegates[0], delegates[1])
+
+	// There are no ENVs for this provider argument
+
+	context := map[string]interface{}{
+		"random_suffix":                         acctest.RandString(t, 10),
+		"impersonate_service_account_delegates": delegatesString,
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		// No PreCheck for checking ENVs
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSdkProvider_impersonate_service_account_delegates_testProvisioning(context),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.google_provider_config_sdk.default", "impersonate_service_account_delegates.#", "2"),
+				),
+			},
+		},
+	})
+}
+
+func testAccSdkProvider_impersonate_service_account_delegates_emptyListUsage(t *testing.T) {
+	acctest.SkipIfVcr(t) // Test doesn't interact with API
+
+	context := map[string]interface{}{
+		"random_suffix":                         acctest.RandString(t, 10),
+		"impersonate_service_account_delegates": "[]",
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		// No PreCheck for checking ENVs
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSdkProvider_impersonate_service_account_delegates_testProvisioning(context),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.google_provider_config_sdk.default", "impersonate_service_account_delegates.#", "0"),
+				),
+			},
+		},
+	})
+}
+
+func testAccSdkProvider_impersonate_service_account_delegates_usage(t *testing.T) {
+	acctest.SkipIfVcr(t) // Test doesn't interact with API
+
+	context := map[string]interface{}{
+		"random_suffix": acctest.RandString(t, 10),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		// No PreCheck for checking ENVs
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSdkProvider_impersonate_service_account_delegates_testViaFailure_1(context),
+			},
+			{
+				// This needs to be split into a second step as impersonate_service_account_delegates does
+				// not tolerate unknown values
+				Config:      testAccSdkProvider_impersonate_service_account_delegates_testViaFailure_2(context),
+				ExpectError: regexp.MustCompile("Error creating Topic: googleapi: Error 403: User not authorized"),
```

Comment:
See https://github.com/GoogleCloudPlatform/magic-modules/pull/11785#issuecomment-2371775320 for a link to a test failing this way

Reply:
Another factor could be whether the Service Account Credentials API is enabled, but this acceptance test doesn't use any newly-provisioned projects. I expect the long-lived test project that the test uses has the necessary APIs enabled already.

Reply:
`IAM Service Account Credentials API` was not enabled on the VCR project, I've just enabled it now

Reply:
Thanks for spotting/sorting that! I'll run the test in TeamCity to double check if it passes now.

Reply:
[SDK test passed ok!](https://hashicorp.teamcity.com/buildConfiguration/TerraformProviders_GoogleCloud_GOOGLE_BETA_MMUPSTREAMTESTS_GOOGLEBETA_PACKAGE_PROVIDER/242443?hideTestsFromDependencies=false&hideProblemsFromDependencies=false&expandBuildChangesSection=true&expandBuildDeploymentsSection=false) That API is enabled in the testing project I was using for local development of this PR, so the unexplained test failure seems to just be due to that API.

There isn't any impact on the plugin-framework test, as currently there isn't a `""impersonate_service_account_delegates controls which service account is used for actions"` test case. I'd like to wait until there are non-Firebase resources implemented with the PF first before adding that test case, because the config for using impersonate_service_account_delegates is already pretty long and that impacts how understandable/maintainable the test case is.

Reply:
makes sense! glad this test worked though.

Reply:
Ah, I realised I accidentally ran 0 tests in that last build - the SDK test is misnamed and I entered the 'correct' name.